lib/chef/resource/homebrew_cask.rb

```diff
@@ -34,7 +34,7 @@ class HomebrewCask < Chef::Resource
 
       property :cask_name, String,
         description: "An optional property to set the cask name if it differs from the resource block's name.",
-        regex: %r{^[\w/-]+$},
+        regex: %r{^[\w/\-@]+$},
```

Comment:
i've got a mild concern here which is i don't understand what the behavior of the `-` character was before this change.  i'm surprised it was legal at all and didn't throw a syntax error.

Reply:
does a trailing `-` behave like `\-` instead of being a character range metacharacter?

Reply:
I asked some rubyist, they said they can't find any official document or tests on ruby but one person finds oniguruma library test case like old code.

The '-' of last character in the [] match '-'. This is Ruby's regular expression library's expected behavior.

https://github.com/k-takata/Onigmo/blob/dd8a18af5c2f2871104b1bdbf3bbb597ec9e4665/testc.c#L217

Reply:
TIL